velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -847,4 +854,129 @@ core::TypedExprPtr extractFiltersFromRemainingFilter(
   return expr;
 }
 
+#ifdef VELOX_ENABLE_PARQUET
+namespace {
+
+std::optional<TimestampUnit> getTimestampUnit(
+    const config::ConfigBase& config,
+    const char* configKey) {
+  if (const auto unit = config.get<uint8_t>(configKey)) {
+    VELOX_CHECK(
+        unit == 0 /*second*/ || unit == 3 /*milli*/ || unit == 6 /*micro*/ ||
+            unit == 9 /*nano*/,
+        "Invalid timestamp unit: {}",
+        unit.value());
+    return std::optional(static_cast<TimestampUnit>(unit.value()));
+  }
+  return std::nullopt;
+}
+
+std::optional<std::string> getTimestampTimeZone(
+    const config::ConfigBase& config,
+    const char* configKey) {
+  if (const auto timezone = config.get<std::string>(configKey)) {
+    return timezone.value();
+  }
+  return std::nullopt;
+}
+
+} // namespace
+#endif
+
+void updateWriterOptionsFromHiveConfig(
+    dwio::common::FileFormat fileFormat,
+    const std::shared_ptr<const HiveConfig>& hiveConfig,
+    const config::ConfigBase* sessionProperties,
+    std::shared_ptr<dwio::common::WriterOptions>& writerOptions) {
+  if (fileFormat == dwio::common::FileFormat::PARQUET) {
+#ifdef VELOX_ENABLE_PARQUET
```

Comment:
Can you add two anonymous functions in cc to handle parquet and dwrf file formats separately based on the file format? Thanks!